lapis/README.md

```diff
@@ -91,6 +91,30 @@ For example:
 ./gradlew bootRun --args='--silo.url=http://localhost:8091 --lapis.databaseConfig.path=../lapis-e2e/testData/singleSegmented/testDatabaseConfig.yaml --referenceGenomeFilename=../lapis-e2e/testData/singleSegmented/reference_genomes.json  --server.port=8090'
 ```
 
+If running from scratch you need to pre-build the code!!!
+```
+./gradlew build bootRun --args='--silo.url=http://localhost:8091 --lapis.databaseConfig.path=../lapis-e2e/testData/singleSegmented/testDatabaseConfig.yaml --referenceGenomeFilename=../lapis-e2e/testData/singleSegmented/reference_genomes.json --server.port=8096'
+```
+
+## ANTLR grammar
+
+
+LAPIS uses [ANTLR](https://www.antlr.org/) as a parser generator for variant and advanced queries. The grammar can be found in `src/main/antlr`. When the package is built you can find the produced query parser, lexer and listener modules in `lapis/build/generated-src/antlr/...`.
+
+To test the ANTLR parser you can run ANTLR locally on your grammar e.g. on the AdvancedQuery.g4 grammar:
+
+```
+antlr4 AdvancedQuery.g4 -o antlr-gen -visitor 
+cd antlr-gen    
+
+javac *.java
+javac -cp ".:/path/to/antlr-4.13.2-complete.jar" *.java
+// see the tokens - paste expression and then hit Crtl+D
+java -cp ".:/path/to/antlr-4.13.2-complete.jar" org.antlr.v4.gui.TestRig AdvancedQuery start -tokens
+// see the semantic tree - paste expression and then hit Crtl+D
+java -cp ".:/path/to/antlr-4.13.2-complete.jar" org.antlr.v4.gui.TestRig AdvancedQuery start -tree
+```
```

Comment:
We could instead also just say: Build the code (e.g. by executing tests) and check the compiled output. Since we have antrl integrated in the gradle build. Or is there an advantage of running antlr manually over running it via gradle?

Reply:
yes - this way you can actually see the tokens and the tree that antlr produces on a given query this is very useful for initially writing your grammar and later debugging purposes

Reply:
Ok, fine by me :+1: